### PR TITLE
Filter duplicate ips or hostnames for ingress

### PR DESCRIPTION
**Pull request by FengyunPan**
Fix issue: #48654

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/printers/internalversion/printers.go [hunk 1]
```diff
@@ -672,7 +672,18 @@ func loadBalancerStatusStringer(s api.LoadBalancerStatus, wide bool) string {
 			result = append(result, ingress[i].Hostname)
 		}
 	}
-	r := strings.Join(result, ",")
+
+	// Remove duplicate item
```
Comment (shiywang) [inline pkg/printers/internalversion/printers.go]: can you move those code into a function ? also I'm not in favor of this deduce after sort, how about this one: https://play.golang.org/p/GG2RQ1ot6t. btw do we have some exist function for this ? cc @alexandercampbell

**Comment by shiywang** [comment]
@kubernetes/sig-cli-pr-reviews

File: pkg/printers/internalversion/printers.go (hunk 1 above)
Comment (FengyunPan) [inline pkg/printers/internalversion/printers.go]: @shiywang Yeah, thank for your review, and how about "sets.NewString()"?

Comment (shiywang) [inline pkg/printers/internalversion/printers.go]: @FengyunPan that's even better, thanks

File: pkg/printers/internalversion/printers.go [hunk 2]
```diff
@@ -664,15 +664,16 @@ func printCronJobList(list *batch.CronJobList, options printers.PrintOptions) ([
...
 	ingress := s.Ingress
-	result := []string{}
+	result := sets.NewString()
 	for i := range ingress {
 		if ingress[i].IP != "" {
-			result = append(result, ingress[i].IP)
+			result.Insert(ingress[i].IP)
 		} else if ingress[i].Hostname != "" {
-			result = append(result, ingress[i].Hostname)
+			result.Insert(ingress[i].Hostname)
```
Comment (alexandercampbell) [inline pkg/printers/internalversion/printers.go]: Would it make sense to print a warning message if duplicates are filtered?

Comment (FengyunPan) [inline pkg/printers/internalversion/printers.go]: Yes, of course.

**Comment by FengyunPan** [comment]
ping @adohe

**Comment by xiangpengzhao** [comment]
Testing this flake: #49824

File: pkg/printers/internalversion/printers.go (hunk 2 above)
Comment (alexandercampbell) [inline pkg/printers/internalversion/printers.go]: Will you add that to this PR?

Comment (FengyunPan) [inline pkg/printers/internalversion/printers.go]: @alexandercampbell Sorry, after checking the code again, the duplicate IPs are passed by ingress, but 'kubectl get' doesn't filter it, my PR adds codes  to filter it and make it simpler to user. So I think that printing a warning message is unnecessary, do think so? If not, I will add a warning meesage.

Comment (alexandercampbell) [inline pkg/printers/internalversion/printers.go]: I see. Yep, the way it is looks fine to me. Thanks for following up!

**Pull request merged**